app/models/concerns/parameterizable.rb

```diff
@@ -27,7 +27,7 @@ def to_param
       end
 
       def self.from_param(id_name)
-        self.find_by_id(id_name.to_i)
+        self.find_by_id(id_name.to_i) if id_name =~ /\A\d+-/
```

Comment:
Only find by id if it is in the correct format - if the parameter is for example "1model", it will find the record with ID 1 even though it isn't the one we are looking for.